curl-websocket/curl-websocket.c

```diff
@@ -775,20 +787,29 @@ static size_t _cws_process_frame(struct cws_data *priv, const char *buffer, size
 static size_t _cws_receive_data(const char *buffer, size_t count, size_t nitems, void *data) {
     struct cws_data *priv = data;
     size_t len = count * nitems;
+
+    if (priv->redirection) {
+        return len;
+    }
+
     while (len > 0) {
         size_t used = _cws_process_frame(priv, buffer, len);
         len -= used;
         buffer += used;
     }
 
-    return count * nitems;
+    return len;
```

Comment:
you can't use `len` here as it's being decreased in the loop (`while (len > 0)`)